### ISSUE TITLE
Is there a async router function?

### ISSUE THREAD

**Issue by dzz9143** [chimurai/http-proxy-middleware#382]
### Question
I want to dynamically change target for proxy, `router options`  as a function seems to be a good choice, but it's a sync function returns a target string.

Is there a way to do the async router function, something like this
```javascript
async function asyncRouterFunc(req) {
    const target = await getNewTargetBasedOnReq(req);
    return target;
}
```

**Comment by chimurai** [comment]
Just merged this in PR #379 

I released a beta version:
`npm install http-proxy-middleware@0.21.0-beta.1`

Let me know if you run into issues.

**Comment by hjagodzinski** [comment]
@chimurai it works like a charm. Thanks!

**Comment by chimurai** [comment]
published: `http-proxy-middleware@0.21.0` with async router support
